docs/src/jekyll/docs.md

```diff
@@ -0,0 +1,562 @@
+
+```scala
+val fch: Fetch[(User, User)] = for {
+  aUser <- getUser(1)
+  anotherUser <- getUser(aUser.id + 1)
+} yield (aUser, anotherUser)
+
+val result: (User, User) = Fetch.run[Id](fch)
+// Fetching users List(1)
+// Fetching users List(2)
+
+//=> result: (User, User) = (User(1,@egg_1),User(2,@egg_2))
+```
+
+### Batching
+
+If we combine two independent requests to the same data source, Fetch will
+automatically batch them together into a single request. Fetch's `join` operator
+help us tell the library that two fetches are independent, and thus can be batched
+if they use the same data source:
+
+```scala
+import cats.syntax.cartesian._
+
+val fch: Fetch[(User, User)] = getUser(1).product(getUser(2))
+
+val result: (User, User) = Fetch.run[Id](fch)
+// Fetching users List(1, 2)
+
+//=> result: (User, User) = (User(1,@egg_1),User(2,@egg_2))
+```
+
+### Deduplication
+
+If two independent requests ask for the same identity, Fetch will detect that and
+deduplicate such id.
+
+```scala
+val fch: Fetch[(User, User)] = getUser(1).product(getUser(1))
+
+val result: (User, User) = Fetch.run[Id](fch)
+// Fetching users List(1)
+
+//=> result: (User, User) = (User(1,@egg_1),User(1,@egg_1))
+```
+
+### Caching
+
+During a fetch, previously requested results are implicitly cached. This allows us to write
+fetches in a very modular way, asking for all the data they need as if it
+was in memory; furthermore, it also avoids refetching an identity that may have changed
+during the course of a fetch execution, which can lead to inconsistencies in the data.
+
+```scala
+val fch: Fetch[(User, User)] = for {
+  aUser <- getUser(1)
+  anotherUser <- getUser(1)
+} yield (aUser, anotherUser)
+
+val result: (User, User) = Fetch.run[Id](fch)
+// Fetching users List(1)
+
+//=> result: (User, User) = (User(1,@egg_1),User(1,@egg_1))
+```
+
+As you can see, the `User` with id 1 was fetched only once in a single round-trip. The next
+time it was needed we used the cached versions, thus avoiding another request to the user data
+source.
+
+## Combining data from multiple sources
+
+Now that we know about some of the optimizations that Fetch can make to read data efficiently,
+let's look at how we can combine more than one data source. Imagine that we are rendering a blog
+and have the following types for posts and post information:
+
+```scala
+type PostId = Int
+case class Post(id: PostId, author: UserId, content: String)
+case class PostInfo(topic: String)
+```
+
+As you can see, every `Post` has an author, but it refers to it by its id. We'll implement two data sources:
+
+- one for retrieving a post given a post id
+- another for retrieveng post metadata given a post id
+
+```scala
+implicit object PostSource extends DataSource[PostId, Post]{
+  override def fetch(ids: List[PostId]): Eval[Map[PostId, Post]] = {
+    Eval.later({
+      println(s"Fetching posts $ids")
+      ids.map(i => (i, Post(i, i % 3 + 1, s"An article with id $i"))).toMap
```

Comment:
I think this use of map to generate data is confusing. I think each datasource should have a list, vector or map call Database or whatever that illustrates that the data may be elsewhere and the Datasource is nothing but an access layer on top of it.